src/neural/metal/metal_common.h

```diff
@@ -0,0 +1,88 @@
+/*
+  This file is part of Leela Chess Zero.
+  Copyright (C) 2021 The LCZero Authors
+
+  Leela Chess is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  Leela Chess is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with Leela Chess.  If not, see <http://www.gnu.org/licenses/>.
+
+  Additional permission under GNU GPL version 3 section 7
+
+  If you modify this Program, or any covered work, by linking or
+  combining it with NVIDIA Corporation's libraries from the NVIDIA CUDA
+  Toolkit and the NVIDIA CUDA Deep Neural Network library (or a
+  modified version of those libraries), containing parts covered by the
+  terms of the respective license agreement, the licensors of this
+  Program grant you additional permission to convey the resulting work.
+*/
+#pragma once
+
+namespace lczero {
+namespace metal_backend {
+
+static int kNumOutputPolicy = 1858;
+static int kInputPlanes = 112;
+
+struct InputsOutputs {
+  InputsOutputs(int maxBatchSize, bool wdl, bool moves_left, bool conv_policy) {
+    input_masks_mem_ = (uint64_t*)malloc(maxBatchSize * kInputPlanes * sizeof(uint64_t));
```

Comment:
Any reason why it's not just std::vectors?  
Even if vectors are for some reason not fine, then it can be `std::unique_ptr<uint64_t[]>` etc.

And if std is for some reason not available, it's better to have a helper container class which will handle allocation, rather than explicitly malloc/free for every pointer. (and also why malloc/free rather than new/delete?)

Reply:
If it's okay @mooskagh, I can just make all these into std::vectors, but considering I took this from the onednn backend, does this mean we would need to replace those as well?

Reply:
I can take care of onednn later, there are some pending changes that I don't want to have to rebase.